controlplane/kubeadm/api/v1beta1/kubeadm_control_plane_types.go

```diff
@@ -234,6 +239,20 @@ type RemediationStrategy struct {
 	MinHealthyPeriod *metav1.Duration `json:"minHealthyPeriod,omitempty"`
 }
 
+// MachineNamingStrategy allows changing the naming pattern used when creating Machines.
+// InfraMachines & KubeadmConfigs will use the same name as the corresponding Machines.
+type MachineNamingStrategy struct {
+	// Template defines the template to use for generating the names of the Machine objects.
+	// If not defined, it will fallback to `{{ .kubeadmcontrolplane.name }}-{{ .random }}`.
+	// If the templated string exceeds 63 characters, it will be trimmed to 58 characters and will
+	// get concatenated with a random suffix of length 5.
+	// The templating mechanism provides the following arguments:
+	// * `.kubeadmcontrolplane.name`: The name of the KubeadmControlPlane object.
```

Comment:
I think we should provide also cluster name

Reply:
Okay for me

Reply:
I have added clustername too.